doc/src/pair_mliap.rst

```diff
@@ -167,6 +173,34 @@ to specify the path for these *model* and *descriptor* files.
    larger systems since their size depends on the total number of
    neighbors per MPI process.
 
+The *unified* keyword is followed by an argument specifying the
+filename containing the serialized unified Python object and the "ghostneigh" toggle
+(0/1) to disable/enable the construction of neighbors lists including
+neighbors of ghost atoms. If the filename ends in '.pt', or '.pth', it will be loaded
+using pytorch; otherwise, it will be loaded as a pickle file.
+If ghostneigh is enabled, it is recommended to set :doc:`comm_modify <comm_modify>`
```

Comment:
FYI: when I enable the "ghostneigh" flag in the `in.mliap.unified.lj.Ar` example, I get a segmentation fault. Is this expected behavior? Can this be checked for and an error triggered?
Increasing the communication cutoff as suggested below did not make a difference.

Reply:
I don't believe this is expected behavior. The error likely comes from neighbor list memory allocation as this segfault occurs while calculating Per MPI Rank memory allocation (see backtrace below). If I reduce the problem size from 4000 atoms to 256 I get a neighbor list overflow instead of a segfault.
```
Core was generated by `lmp -i in.mliap.unified.lj.Ar'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000149cd25883d8 in int_mallinfo () from /lib64/libc.so.6
Missing separate debuginfos, use: yum debuginfo-install glibc-2.28-151.el8.x86_64
(gdb) bt
#0  0x0000149cd25883d8 in int_mallinfo () from /lib64/libc.so.6
#1  0x0000149cd258e09c in mallinfo () from /lib64/libc.so.6
#2  0x0000149cd32247c6 in LAMMPS_NS::Info::get_memory_info (this=this@entry=0x7ffc14762a90, meminfo=meminfo@entry=0x7ffc14762a20)
    at /projects/ml4chem/sanaya/lammps/src/info.cpp:1300
#3  0x0000149cd3310b6b in LAMMPS_NS::Output::memory_usage (this=0x744a30) at /projects/ml4chem/sanaya/lammps/src/output.cpp:1018
#4  0x0000149cd33110d1 in LAMMPS_NS::Output::setup (this=0x744a30, memflag=memflag@entry=1) at /projects/ml4chem/sanaya/lammps/src/output.cpp:299
#5  0x0000149cd33f3ae2 in LAMMPS_NS::Verlet::setup (this=0x73ca20, flag=1) at /projects/ml4chem/sanaya/lammps/src/verlet.cpp:160
#6  0x0000149cd33a0c6f in LAMMPS_NS::Run::command (this=0x91bbc0, narg=1, arg=0x749180) at /projects/ml4chem/sanaya/lammps/src/run.cpp:171
#7  0x0000149cd323d1b8 in LAMMPS_NS::Input::execute_command (this=0x72d6f0) at /projects/ml4chem/sanaya/lammps/src/input.cpp:847
#8  0x0000149cd323d58d in LAMMPS_NS::Input::file (this=0x72d6f0) at /projects/ml4chem/sanaya/lammps/src/input.cpp:302
#9  0x0000000000400e85 in main (argc=<optimized out>, argv=<optimized out>) at /projects/ml4chem/sanaya/lammps/src/main.cpp:76
```

Reply:
The gdb trace is bogus. Something has written to an allocated storage block beyond the allocated size and thus messed up the internal malloc data structures.

I have run the code with valgrind and identified the location of the problem at this location:
https://github.com/lammps/lammps/pull/3458/files#diff-cb6b51f0e9ebcb4ecda7e8a34b450fb05f24b7de8a8bfbac614cbfd21af03b6dR256

The index `i` overflows the dimension of the `data->eatoms` array which is of size `data->nlistatoms`.
I am implementing a hack that avoids the crash, but will likely lead to incorrect per-atom energy data.
I would expect that `data->eatoms` would need to be allocated (and cleared) so it includes ghost atoms and then a reverse communication would be needed to sum the contributions to ghost atoms to their principal owned atoms.

Reply:
Also found and fixed a memory leak, an initialization issue and noticed that destructors of some polymorph classes were not make virtutal and general our conventions for using virtual/override were not fully followed.

Reply:
@Boogie3D I think I have sorted this out for the LJ example. I could not test the hippynn examples, since I did not see any of the .pt files.
I also added a test that creates two LAMMPS instances with mliap.unified.lj and lj/cut pair styles, respectively and check some results after a couple of MD steps.

Reply:
> @Boogie3D I think I have sorted this out for the LJ example. I could not test the hippynn examples, since I did not see any of the .pt files. I also added a test that creates two LAMMPS instances with mliap.unified.lj and lj/cut pair styles, respectively and check some results after a couple of MD steps.

The scripts to create the hippynn .pt files are in the hippynn [repo](https://github.com/Boogie3D/hippynn/tree/lammps_mliap/examples/lammps) but unfortunately require HIPNN training to be performed first; the .pt files themselves are not uploaded anywhere.

Thank you for your various improvements and for solving the problem with memory allocation and LJ so quickly. I am beginning graduate school soon and am not currently working at LANL so I may not have had the time to do so myself.

Reply:
> the .pt files themselves are not uploaded anywhere

Can you send me a copy of those files via email? I would very much like to make certain that my changes did not break anything.